irregex.scm

```diff
@@ -92,11 +92,14 @@
 (define (irregex-lengths x) (vector-ref x 6))
 (define (irregex-names x) (vector-ref x 7))
 
-(define (vector-copy v)
-  (let ((r (make-vector (vector-length v))))
-    (do ((i (- (vector-length v) 1) (- i 1)))
-        ((< i 0) r)
-      (vector-set! r i (vector-ref v i)))))
+(cond-expand
```

Comment:
The raw irregex.scm is meant to load in any R5RS system, which means no cond-expand.

For larceny there's now an R7RS library which should work.  If you do want to cond-expand to use the native vector-copy you can do it in irregex.sld.